src/charm.py

```diff
@@ -255,10 +264,16 @@ def _on_remove(self, _event) -> None:
         """Uninstall the Grafana Agent snap."""
         self.unit.status = MaintenanceStatus("Uninstalling grafana-agent snap")
         try:
-            self.snap.ensure(state=snap.SnapState.Absent)
-        except snap.SnapError as e:
+            self.snap.ensure(state=snap_lib.SnapState.Absent)
+        except snap_lib.SnapError as e:
             raise GrafanaAgentInstallError("Failed to uninstall grafana-agent") from e
 
+    def _on_upgrade_charm(self, _event=None):
```

Comment:
[nit] I normally only prefix `_` to args in case they're not being used.

Reply:
Agreed. How about dropping the default as well?:

Reply:
yeah this was a mistake.  fixed

Reply:
removed the default as well